File: app/src/main/java/com/example/tourethiopia/hotels/HotelDetailActivity.java
package com.example.tourethiopia.hotels;

import android.support.v4.content.ContextCompat;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.tourethiopia.R;
import com.example.tourethiopia.place.PlacesData;

public class HotelDetailActivity extends AppCompatActivity {

    public static final String EXTRA_HOTELS_ID="hotelId";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_hotel_detail);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbarid);
        setSupportActionBar(toolbar);
        ActionBar actionBar= getSupportActionBar();
        actionBar.setDisplayHomeAsUpEnabled(true);

        int hotelId = (Integer)getIntent().getExtras().get(EXTRA_HOTELS_ID);
        int hotelName = HotelsData.HOTELS_DATA[hotelId].getName();
        int addressDesc= HotelsData.HOTELS_DATA[hotelId].getAddress();
        int hotelPhone = HotelsData.HOTELS_DATA[hotelId].getPhonenumber();
        int hotelImage = HotelsData.HOTELS_DATA[hotelId].gethotelimagersourceid();

        TextView textView1 = (TextView)findViewById(R.id.hotel_text);
        textView1.setText(hotelName);
        TextView textView2 = (TextView)findViewById(R.id.hoteladdress);
        textView2.setText(addressDesc);
        TextView textView3 = (TextView)findViewById(R.id.hotel_phonenumber);
        textView3.setText(hotelPhone);
        ImageView imageView = (ImageView)findViewById(R.id.hoteldetail_image);
        imageView.setImageDrawable(ContextCompat.getDrawable(this, hotelImage));
    }
}

File: app/src/main/java/com/example/tourethiopia/hotels/HotelDetailActivity.kt
package com.example.tourethiopia.hotels

import android.os.Bundle
import android.support.v4.content.ContextCompat
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.example.tourethiopia.R

class HotelDetailActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_hotel_detail)
        val toolbar = findViewById<View>(R.id.toolbarid) as Toolbar
        setSupportActionBar(toolbar)
        val actionBar = supportActionBar
        actionBar!!.setDisplayHomeAsUpEnabled(true)
        val hotelId = intent.extras!![EXTRA_HOTELS_ID] as Int
        val hotelName: Int = HotelsData.Companion.HOTELS_DATA.get(hotelId).name
        val addressDesc: Int = HotelsData.Companion.HOTELS_DATA.get(hotelId).address
        val hotelPhone: Int = HotelsData.Companion.HOTELS_DATA.get(hotelId).phonenumber
        val hotelImage: Int = HotelsData.Companion.HOTELS_DATA.get(hotelId).gethotelimagersourceid()
        val textView1 = findViewById<View>(R.id.hotel_text) as TextView
        textView1.setText(hotelName)
        val textView2 = findViewById<View>(R.id.hoteladdress) as TextView
        textView2.setText(addressDesc)
        val textView3 = findViewById<View>(R.id.hotel_phonenumber) as TextView
        textView3.setText(hotelPhone)
        val imageView = findViewById<View>(R.id.hoteldetail_image) as ImageView
        imageView.setImageDrawable(ContextCompat.getDrawable(this, hotelImage))
    }

    companion object {
        const val EXTRA_HOTELS_ID = "hotelId"
    }
}